### PR TITLE
GODRIVER-3872: Mark Server version 4.2 as EOL

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-3872

EOL 4.2 support in the driver.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -1985,12 +1981,12 @@ buildvariants:
       - name: ".goleak"
   - matrix_name: "tests-rhel-44-plus-zlib-zstd-support"
     tags: ["pullrequest"]
-    matrix_spec: {version: ["4.2", "4.4", "5.0", "6.0", "7.0", "8.0"], os-ssl-40: ["rhel87-64"]}
+    matrix_spec: {version: ["4.4", "5.0", "6.0", "7.0", "8.0"], os-ssl-40: ["rhel87-64"]}
     display_name: "${version} ${os-ssl-40}"
     tasks:
       - name: ".test !.enterprise-auth !.snappy"
   - matrix_name: "tests-windows-42-plus-zlib-zstd-support"
```
[the remark this answers is not in the document]
Comment (qingyang-hu) [reply .evergreen/config.yml]: We need to address this.

File: internal/spectest/skip.go [hunk 2]
```diff
@@ -1075,6 +1075,23 @@ var skipTests = map[string][]skipCase{
 			},
 		},
 	},
+
+	// TODO(GODRIVER-3872): Mark Server version 4.2 as EOL from driver's PoV
```
Comment (qingyang-hu) [inline internal/spectest/skip.go]: [suggestion] "GODRIVER-3872" is the ticket this PR closes. We may need to refer to a different ticket to prevent confusion.

File: x/mongo/driver/topology/server.go [hunk 3]
```diff
@@ -540,8 +537,8 @@ func (s *Server) ProcessError(err error, describer mnet.Describer) driver.Proces
 		s.RequestImmediateCheck()
 
 		res := driver.ServerMarkedUnknown
-		// If the node is shutting down or is older than 4.2, we synchronously clear the pool
-		if wcerr.NodeIsShuttingDown() || wireVersion == nil || wireVersion.Max < wireVersion42 {
+		// If the node is shutting down or unknown, we synchronously clear the pool.
+		if wcerr.NodeIsShuttingDown() || wireVersion == nil {
```
Comment (qingyang-hu) [inline x/mongo/driver/topology/server.go:541]: [suggestion] We may add a test case to cover the `wireVersion == nil` arm.

**Comment by qingyang-hu** [comment]
[blocking] We need to update: https://github.com/prestonvasquez/mongo-go-driver/blob/godriver-3872/eol-server-42/x/mongo/driver/topology/fsm.go#L23 and correlated tests.

**Comment by qingyang-hu** [comment]
[question] Do we also need to update the comments in "./mongo/"? e.g., https://github.com/prestonvasquez/mongo-go-driver/blob/godriver-3872/eol-server-42/mongo/doc.go#L116 and https://github.com/prestonvasquez/mongo-go-driver/blob/godriver-3872/eol-server-42/mongo/options/clientoptions.go#L739

**Review by qingyang-hu — CHANGES_REQUESTED** [review]

File: x/mongo/driver/topology/server.go (hunk 3 above)
Comment (prestonvasquez) [reply x/mongo/driver/topology/server.go:541]: Suggest following up on another ticket if this isn't already tested. `wireVersion == nil` is not new behavior.

**Comment by prestonvasquez** [comment]
> https://github.com/prestonvasquez/mongo-go-driver/blob/godriver-3872/eol-server-42/mongo/options/clientoptions.go#L739

I think we should leave server references as-is. The resulting comment might read kind of strangely, for example:

> In-use encryption is a new feature in MongoDB 4.2 that allows specific data fields to be encrypted, the drivers minimum version is 4.4.

File: internal/spectest/skip.go (hunk 2 above)
Comment (prestonvasquez) [reply internal/spectest/skip.go]: Good catch 👍

**Review by qingyang-hu — APPROVED** [review]
